sdcm/tester.py

```diff
@@ -787,19 +788,30 @@ def init_resources(self, loader_info=None, db_info=None,
         elif cluster_backend == 'baremetal':
             self.get_cluster_baremetal()
 
-        seeds_num = self.params.get('seeds_num', default=1)
-        if self.params.get('instance_provision') == 'mixed':
-            for node in self.db_cluster.nodes:
-                if seeds_num == 0:
-                    break
-                if node.is_spot:
-                    node.is_seed = True
-                    seeds_num -= 1
+    def set_seeds(self):
+        seeds_selector = self.params.get('seeds_selector')
+        seeds_num = self.params.get('seeds_num')
+
+        if seeds_selector == 'reflector':
+            node = self.db_cluster.nodes[0]
+            node.wait_ssh_up()
+            # When cluster just started, seed IP in the scylla.yaml may be '127.0.0.1'
+            # In this case we want ignore it and wait, when reflector will select real node and update scylla.yaml
+            self.db_cluster.seed_nodes_ips = wait.wait_for(self.db_cluster.get_seed_selected_by_reflector,
```

Comment:
what is the reason that you maintain two places of information about which node is seed?
(db_cluster.seed_nodes_ips and node.is_seed)

Reply:
fixed